### PR TITLE
DFR-969: update solicitor address and contact details

### PR DESCRIPTION
**Pull request by FaisalMoJ**
- Add new endpoint to update solicitor contact details and set the fields null. Also add unit tests for newly created endpoint.
- Add role specific key value pair in JSON files for testing purposes
- Add solicitor roles controller Add solicitor roles controller, service and model classes. Also add data store client interface and unit tests.

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DFR-969

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/client/DataStoreClient.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.SERVICE_AUTHORISATION_HEADER;
+
+@FeignClient(name = "data-store-api", url = "${ccd.data-store.api.baseurl}")
+@Configuration
+public interface DataStoreClient {
+    @GetMapping(
+        path = "/case-users",
+        headers = CONTENT_TYPE + "=" + APPLICATION_JSON_VALUE)
+    CaseAssignedUserRolesResource getCaseAssignedUserRoles(@PathVariable("case_ids") Long caseID, @RequestHeader(HttpHeaders.AUTHORIZATION)
```
Comment (EllisD-B) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/client/DataStoreClient.java:21]: This request variable needs to be a string or list of strings, not a long https://tools.hmcts.net/confluence/display/RCCD/API+Operation%3A+Get+Case-Assigned+Users+and+Roles
To be safe I would always pass it as a list

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/UpdateConsentedCaseController.java [hunk 2]
```diff
@@ -86,6 +90,32 @@ public ResponseEntity<AboutToStartOrSubmitCallbackResponse> updateCase(
...
+        @RequestHeader(value = AUTHORIZATION_HEADER, required = false) String authToken,
+        @RequestBody CallbackRequest ccdRequest) {
+
+        CaseDetails caseDetails = ccdRequest.getCaseDetails();
+        log.info("Received request to update consented case solicitor contact details with Case ID: {}", caseDetails.getId());
+
+        validateCaseData(ccdRequest);
+        Map<String, Object> caseData = caseDetails.getData();
+
+        if (caseData.get(CASE_ROLE).equals(APP_SOLICITOR_POLICY)) {
```
Comment (EllisD-B) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/UpdateConsentedCaseController.java:110]: It's probably wise to set the case role field to null before you exit this callback

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/UpdateContestedCaseController.java [hunk 3]
```diff
@@ -90,6 +101,32 @@ public ResponseEntity<AboutToStartOrSubmitCallbackResponse> updateContestedCase(
...
+        @RequestHeader(value = AUTHORIZATION_HEADER, required = false) String authToken,
+        @RequestBody CallbackRequest ccdRequest) {
+
+        CaseDetails caseDetails = ccdRequest.getCaseDetails();
+        log.info("Received request to update consented case solicitor contact details with Case ID: {}", caseDetails.getId());
+
+        validateCaseData(ccdRequest);
+        Map<String, Object> caseData = caseDetails.getData();
+
+        if (caseData.get(CASE_ROLE).equals(APP_SOLICITOR_POLICY)) {
```
Comment (EllisD-B) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/UpdateContestedCaseController.java:121]: same here

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/client/DataStoreClient.java (hunk 1 above)
Comment (EllisD-B) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/client/DataStoreClient.java:21]: This request variable needs to be a string or list of strings, not a long https://tools.hmcts.net/confluence/display/RCCD/API+Operation%3A+Get+Case-Assigned+Users+and+Roles
To be safe I would always pass it as a list

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/CaseAssignedRoleService.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+public class CaseAssignedRoleService {
+
+    private final DataStoreClient dataStoreClient;
+    private final CaseDataService caseDataService;
+    private final AuthTokenGenerator authTokenGenerator;
+
+    public Map<String, Object> setCaseAssignedUserRole(CaseDetails caseDetails,
+                                                       String authToken) {
+
+        CaseAssignedUserRolesResource resource = dataStoreClient.getCaseAssignedUserRoles(caseDetails.getId(),
```
Comment (EllisD-B) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/CaseAssignedRoleService.java:33]: Related to the change for the data store client, change this to `List.of(caseDetails.getId().toString())`

**Review by EllisD-B — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/client/DataStoreClient.java (hunk 1 above)
Comment (MindGames2015) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/client/DataStoreClient.java:21]: Changed Long caseId, to List of Strings

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/UpdateConsentedCaseController.java (hunk 2 above)
Comment (MindGames2015) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/UpdateConsentedCaseController.java:110]: Set currentUserCaseRole to null, as caseData.put(CASE_ROLE, null), for both consented and contested;

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/CaseAssignedRoleService.java (hunk 4 above)
Comment (MindGames2015) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/CaseAssignedRoleService.java:33]: Modified the method signature, and updated this line of code as well.

**Review by EllisD-B — APPROVED** [review]
nice work!

**Pull request merged**